File: fincapp-backend/src/FincApp.API/DTOs/PayloadMutationsDto.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace FincApp.API.DTOs;

public class PayloadMutationsDto
{
    [JsonPropertyName("animals")]
    public List<SyncAnimalDto> Animals { get; set; } = new();

    [JsonPropertyName("weight_logs")]
    public List<SyncWeightLogDto> WeightLogs { get; set; } = new();

    [JsonPropertyName("health_records")]
    public List<SyncHealthRecordDto> HealthRecords { get; set; } = new();
}

File: fincapp-frontend/mobile-app/app/src/main/java/com/irwi/fincapp/network/dto/PayloadMutationsDto.java
package com.irwi.fincapp.network.dto;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;

public class PayloadMutationsDto {
    public List<SyncAnimalDto> animals = new ArrayList<>();
    @SerializedName("weight_logs") public List<SyncWeightLogDto> weightLogs = new ArrayList<>();
    @SerializedName("health_records") public List<SyncHealthRecordDto> healthRecords = new ArrayList<>();
}
